spi/src/main/java/org/jboss/pnc/spi/datastore/repositories/BuildRecordRepository.java

```diff
@@ -77,12 +79,32 @@ default BuildRecord getLatestSuccessfulBuildRecord(List<BuildRecord> buildRecord
                 .orElse(null);
     }
 
+    BuildRecord getAnyLatestSuccessfulBuildRecordWithBuildConfig(Integer configurationId, boolean temporaryBuild);
+
+    default BuildRecord getAnyLatestSuccessfulBuildRecord(List<BuildRecord> buildRecords, boolean temporaryBuild) {
+        return buildRecords.stream()
+                .filter(record -> record.getStatus() == BuildStatus.SUCCESS)
+                // First part includes temporary BRs and second part includes persistent BRs
+                .filter(record -> (temporaryBuild && record.isTemporaryBuild()) || !record.isTemporaryBuild())
```

Comment:
note: maybe this whole method can be deleted, see other comments
```suggestion
                .filter(record -> temporaryBuild || !record.isTemporaryBuild())
```

Reply:
ack ... maybe less readable at first glance ? but...who will ever forget... and putting a comment will do :-)